gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandler.java

```diff
@@ -61,4 +70,19 @@ public GoodbyeCommandHandler(final InstallationService installationService) {
             return Single.just(new GoodbyeReply(command.getId(), CommandStatus.ERROR));
         }
     }
+
+    private void rejectAllPromotionToValidate() {
+        PromotionQuery promotionQuery = new PromotionQuery();
+        promotionQuery.setStatuses(List.of(PromotionEntityStatus.TO_BE_VALIDATED));
+
+        promotionService
+            .search(promotionQuery, null, null)
+            .getContent()
+            .forEach(
+                promotionEntity -> {
+                    promotionEntity.setStatus(PromotionEntityStatus.REJECTED);
+                    promotionService.createOrUpdate(promotionEntity);
+                }
+            );
+    }
```

Comment:
why not introducing a new method in promotionService which:
- set all local promotions to `REJECTED`
- send a new bridge command to the source environments to pass these promotions to `REJECTED` 

maybe a new feature ?

Reply:
As discussed we will need to handled Cockpit <-> Apim unlink